source/mesh_deformation/fastscapecc.cc

```diff
@@ -0,0 +1,1068 @@
+          double fastscape_timestep_in_years = aspect_timestep_in_years / fastscape_iterations;
+          while (fastscape_timestep_in_years > maximum_fastscape_timestep)
+            {
+              fastscape_iterations *= 2;
+              fastscape_timestep_in_years *= 0.5;
+            }
+
+          std::cout << "[DEBUG] fastscape_steps_per_aspect_step = " << fastscape_steps_per_aspect_step << std::endl;
+          std::cout << "[DEBUG] aspect_timestep_in_years = " << aspect_timestep_in_years << std::endl;
+          std::cout << "[DEBUG] fastscape_iterations = " << fastscape_iterations << std::endl;
+            
+
+// TODO: This is a pretty arbitrary cell, with a pretty arbitrary cell diameter. Is that what you
+// want? If so, perhaps say so in a comment and explain what it is used for.
+//
+// Separately, let's call this variable surface_cell_area in concordance with our convention
+          const double surface_cell_area = surface_mesh.begin_active()->measure();
+
+          // 1. Create the FastScape grid adapter
+// TODO: I haven't looked at the GridAdapter yet, but we should see whether we can't get rid of the const_cast.
+          auto grid = GridAdapterType(const_cast<SurfaceMeshType &>(surface_mesh), surface_cell_area);
+          auto shape = grid.shape();  // Should be {n_grid_nodes}
+
+          // 2. Define the node status array with correct shape and type
+          xt::xtensor<fastscapelib::node_status, 1> node_status_array(shape);
+          std::fill(node_status_array.begin(), node_status_array.end(), fastscapelib::node_status::core);
+          grid.set_nodes_status(node_status_array);
+
+          // // 3. Create the flow graph
+          // auto flow_graph = FlowGraphType(
+          //   grid,
+          //   { fastscapelib::single_flow_router(), fastscapelib::MSTSinkResolver()}
+          // );
+
+          auto flow_graph = FlowGraphType(
+              grid,
+              {
+                  std::make_shared<fastscapelib::single_flow_router>(),
+                  std::make_shared<fastscapelib::mst_sink_resolver>()
+              }
+);
+
+          // 4. Set base level nodes
+          std::vector<std::size_t> base_level_nodes = {0};
+          flow_graph.set_base_levels(base_level_nodes);
+
+          // 5. Build the erosion model
+          fastscapelib::spl_eroder<FlowGraphType> spl_eroder
+          =
+            // fastscapelib::make_spl_eroder(flow_graph, kff, n, m, 1e-5)
+            fastscapelib::make_spl_eroder(flow_graph, 5e-4, 0.4, 1, 1e-5);
+          std::cout << "KFF : " << kff<< std::endl;
+
+          //Only for raster grid 
+          //To propose if we add back the raster grid option later 
+          // auto diffusion_eroder = fastscapelib::make_diffusion_adi_eroder(grid, kdd);
+
+              // Create an xarray with correct shape_vec
+          xt::xarray<double> uplift_rate_vec = xt::zeros<double>({vz.size()});
```

Comment:
Would be safe to have an assertion check that `vz.size()` equals `grid.size()` (I guess it should be equal).